test/common/verify_alert_links_to_SOPs.go

```diff
@@ -0,0 +1,179 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+)
+
+var githubToken = os.Getenv("TOKEN_GITHUB")
+var failedSOPurls = make(chan string)
+var wg sync.WaitGroup
+var countFailedLinks int
+
+func TestSOPUrls(t TestingTB, ctx *TestingContext) {
+
+	if githubToken == "" {
+		t.Skip("Github token not provided, use GITHUB_TOKEN environment variable to specify it")
+	}
+
+	var sopUrls []string
+
+	// test connection to Github API, with single url
+
+	testUrl := "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/README.md"
+	apiUrl := modifyLink(testUrl)
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", apiUrl, nil)
+	if err != nil {
+		log.Fatal("%s", err)
+	}
+
+	req.Header.Add("Accept", `application/json`)
+	req.Header.Add("Authorization", fmt.Sprintf("token %s", githubToken))
+	testResp, err := client.Do(req)
+	if err != nil {
+		t.Log(err)
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+
+		}
+	}(testResp.Body)
+
+	if testResp.StatusCode != 200 {
+		t.Fatal("Response status: ", testUrl, testResp.Status, "Given token does not allow access to SOP URLs")
+	}
+
+	output, err := execToPod("wget -qO - localhost:9090/api/v1/rules",
+		"prometheus-prometheus-0",
+		ObservabilityProductNamespace,
+		"prometheus", ctx)
+	if err != nil {
+		t.Fatal("failed to exec to pod:", err)
+	}
+
+	var ApiOutput prometheusAPIResponse
+
+	err = json.Unmarshal([]byte(output), &ApiOutput)
+	if err != nil {
+		t.Fatal("failed to unmarshal json", err)
+	}
+
+	var rulesResult prometheusv1.RulesResult
+
+	err = json.Unmarshal([]byte(ApiOutput.Data), &rulesResult)
+	if err != nil {
+		t.Fatal("failed to unmarshal json", err)
+	}
+
+	for _, group := range rulesResult.Groups {
+		for _, rule := range group.Rules {
+			switch v := rule.(type) {
+			case prometheusv1.RecordingRule:
+			case prometheusv1.AlertingRule:
+				for annotation, sopUrl := range v.Annotations {
+					if annotation == "sop_url" {
+						sopUrls = append(sopUrls, string(sopUrl))
+					}
+				}
+
+			default:
+				fmt.Printf("Unknown rule type %s", v)
+
+			}
+		}
+	}
+	sopUrls = unique(sopUrls)
+	validateSOPLinks(t, sopUrls)
```

Comment:
Minor thing but you are using `sopUrl` and `sopLink` randomly across the code. I would stick with just one name since it is the same thing. Eg here the the function name is `validateSOPLlinks` but the parameter name is `sopUrls`. I would expect either `validateSopUrls` as function name or `sopLinks` as argument name. Just for the sake of consistency.

Reply:
Done in latest commit